backend/src/main/java/ai/verta/modeldb/experiment/ExperimentServiceImpl.java

```diff
@@ -1128,6 +1128,21 @@ public void getUrlForArtifact(
         throw StatusProto.toStatusRuntimeException(status);
       }
 
+      Map<String, String> projectIdFromExperimentMap =
+          experimentDAO.getProjectIdsByExperimentIds(Collections.singletonList(request.getId()));
+      if (projectIdFromExperimentMap.size() == 0) {
+        errorMessage = "Access is denied. Experiment not found for given id : " + request.getId();
+        ModelDBUtils.logAndThrowError(
```

Comment:
this should be a not found? you are not checking for privileges here

Reply:
Fixed